### ISSUE TITLE
Second order stochastic dominance

### ISSUE THREAD

**Issue by rspadim** [jankrepl/deepdow#113]
Could be possible use deepdow yo second order dominance portfolios?

www.kybernetika.cz/content/2008/2/243/paper.pdf

https://www.wiwi.uni-konstanz.de/typo3temp/secure_downloads/85535/0/e1c9300496b7624a1c5f62720b94796f91d46780/2014-5-19_Improved_Portfolio_Choice_using_Second-order_Stochastic_Dominance_Version87.pdf

https://ampl.com/MEETINGS/TALKS/2013_07_Bergamo_Wed.pdf

**Comment by jankrepl** [comment]
Thank you for the post. I actually never heard of this SSD concept. Really interesting. 

If I understood it correctly, one can characterize (sufficient + necessary condition) the SSD efficient portfolio as a solution of some convex optimization problem, right? 

If that is the case, you can use `cvxpylayers` to define and solve the problem. For some inspiration, check out the `NumericalMarkowitz` implementation [here](https://github.com/jankrepl/deepdow/blob/48484f99aa36863b15fb1ae685659841ce37fe25/deepdow/layers/allocate.py#L185)

So in short, I think it should be possible to use it inside of `deepdow`!

**Comment by jankrepl** [comment]
Closing due to inactivity
